File: June 2022/June_12.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> hs;
        int sum = 0, mx = 0, n = nums.size(), j = 0;
        for(int i = 0; i < n; i++){
            while(hs.find(nums[i]) != hs.end()){
                sum -= nums[j];
                hs.erase(nums[j++]);
            }
            sum += nums[i];
            hs.insert(nums[i]);
            mx = max(mx, sum);
        }
        return mx;
    }
};

File: June 2022/June_12.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int sum = 0, max = 0, j = 0, n = nums.length;
        for(int i = 0; i < n; i++){
            while(set.contains(nums[i])){
                sum -= nums[j];
                set.remove(nums[j++]);
            }
            sum += nums[i];
            set.add(nums[i]);
            max = Math.max(max, sum);
        }
        return max;
    }
}
